cornice/validators/_colander.py

```diff
@@ -0,0 +1,63 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+from webob.multidict import MultiDict
+from colander import Invalid
+
+
+class SchemaError(Exception):
+    pass
+
+
+def validator(schema_class):
+    schema = schema_class()
+
+    def _validate(request):
+
+        if request.content_type == 'application/x-www-form-urlencoded':
+            body = request.POST.mixed()
+        elif not request.content_type == 'application/json':
+            body = request.body
+        else:
+            try:
+                body = request.json_body
+            except ValueError as e:
+                request.errors.add('body', '', 'Invalid JSON')
+                return
+            else:
+                if not hasattr(body, 'items'):
+                    request.errors.add('body', '', 'Should be a JSON object')
+                    return
+
+        cstruct = {'method': request.method,
+                   'url': request.url,
+                   'path': request.path,
+                   'body': body}
+
+        for sub, attr in (('query', 'GET'),
+                          ('headers', 'headers'),
+                          ('cookies', 'cookies')):
+            data = getattr(request, attr)
+            if isinstance(data, MultiDict):
+                data = data.mixed()
+            else:
+                data = dict(data)
+            cstruct[sub] = data
+
+        try:
+            deserialized = schema.deserialize(cstruct)
+            request.validated.update(deserialized)
+        except Invalid as e:
+            translate = request.localizer.translate
+            error_dict = e.asdict(translate=translate)
+            for name, msg in error_dict.items():
+                prefixed = name.split('.')
+                if len(prefixed) != 2:
```

Comment:
if field error is in `body.data.person.age`, shouldn't we set `field` to `data.person.age` ?